sympy/core/mul.py

```diff
@@ -1545,37 +1564,48 @@ def _eval_is_extended_negative(self):
         return self._eval_pos_neg(-1)
 
     def _eval_is_odd(self):
-        is_integer = self.is_integer
```

Comment:
Wouldn't it make sense to keep this to avoid potentially have to evaluate it twice?

Reply:
OK!